### DISCUSSION TITLE
Does latest theia support vscode.executeDocumentRenameProvider command ?

### DISCUSSION THREAD

**Discussion by badrihcl** [eclipse-theia/theia#12274] [Q&A]
We have a vscode extension which uses vscode.executeDocumentRenameProvider. When we install that in theia, we get the error  Command with id 'vscode.executeDocumentRenameProvider' is not registered

**Comment by vince-fugnitto** [comment]
@badrihcl thank you for the discussion, the `vscode.executeDocumentRenameProvider` command is not yet implemented.

**Reply by badrihcl** [reply]
Thanks for the response. Is it planned to be implemented in future? Do you know of any alternative function that I can use to make it work in theia ?

**Reply by msujew** [reply]
@badrihcl These commands are usually implemented on a per-demand basis. You're welcome to contribute to the Theia codebase if you're interested in having this fixed. The implementation for vscode specific commands is done [in this file](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts).
